src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/azuredevops/AzureDevOpsServerPullRequestDecorator.java

```diff
@@ -0,0 +1,414 @@
+                    LOGGER.trace(String.format("line: %d (%s)", line, azureThread.getThreadContext().getRightFileStart().getLine() == locate.getTextRange().getEndLine()));
+
+                    // Check if thread already exists and close thread if issue is closed
+                    if (checkAzureThread(issue, filePath, azureThread, azurePullRequestDetails)) {
+                        threadExists = true;
+                        break;
+                    }
+                }
+                if (issue.getIssue().getStatus() != Issue.STATUS_OPEN) {
+                    LOGGER.info(String.format("SKIPPED ISSUE: Issue status is %s", issue.getIssue().getStatus()));
+                    return;
+                }
+
+                if (threadExists || issue.getIssue().getStatus() != Issue.STATUS_OPEN) {
+                    LOGGER.info(String.format("SKIPPED ISSUE: %s %nFile: %s %nLine: %d %nIssue is already exist in azure",
+                            issue.getIssue().getMessage(),
+                            filePath,
+                            line));
+                    return;
+                }
+
+                String message = String.format("%s: %s ([rule](%s))%n%n[See in SonarQube](%s)",
+                        issue.getIssue().type().name(),
+                        issue.getIssue().getMessage(),
+                        analysisDetails.getRuleUrlWithRuleKey(issue.getIssue().getRuleKey().toString()),
+                        analysisDetails.getIssueUrl(issue.getIssue().key())
+                );
+
+                CommentThread thread = new CommentThread(filePath, locate, message);
+                LOGGER.info(String.format("Creating thread: %s", new ObjectMapper().writeValueAsString(thread)));
+                sendPost(
+                        getThreadApiUrl(azurePullRequestDetails),
+                        new ObjectMapper().writeValueAsString(thread),
+                        "Thread created successfully",
+                        azurePullRequestDetails.getAuthorizationHeader()
+                );
+            } catch (Exception e) {
+                LOGGER.error("Could not create thread on AzureDevOps Server", e);
+                throw new IllegalStateException("Could not create thread on AzureDevOps Server", e);
+            }
+        }
+    }
+
+    private boolean checkAzureThread(PostAnalysisIssueVisitor.ComponentIssue issue, String filePath, CommentThread azureThread,
+                                     AzurePullRequestDetails azurePullRequestDetails) {
+        try {
+            if (azureThread.getThreadContext().getFilePath().equals(filePath)
+                    && azureThread.getComments()
+                    .stream()
+                    .filter(c -> c.getContent().contains(issue.getIssue().key()))
+                    .count() > 0 ) {
+
+                if(issue.getIssue().getStatus() != Issue.STATUS_OPEN
```

Comment:
I think you should use equals(...), because issue.getIssue().getStatus() is not enum but string :(

And please add space between if and (

Reply:
Fixed.